File: wsl-tamer-tauri/src/services/profiles.ts
// Profile Service - Tauri IPC wrapper

import { invoke } from '@tauri-apps/api/core';
import type { WslProfile, AutomationRule, AppConfig } from '../types';

export const profileService = {
  // Profiles
  async getProfiles(): Promise<WslProfile[]> {
    return invoke('get_profiles');
  },

  async getProfile(id: string): Promise<WslProfile | null> {
    return invoke('get_profile', { id });
  },

  async getCurrentProfile(): Promise<WslProfile | null> {
    return invoke('get_current_profile');
  },

  async saveProfile(profile: WslProfile): Promise<void> {
    return invoke('save_profile', { profile });
  },

  async deleteProfile(id: string): Promise<void> {
    return invoke('delete_profile', { id });
  },

  async setDefaultProfile(id: string): Promise<void> {
    return invoke('set_default_profile', { id });
  },

  async applyProfile(id: string): Promise<void> {
    return invoke('apply_profile', { id });
  },

  // Automation rules
  async getRules(): Promise<AutomationRule[]> {
    return invoke('get_automation_rules');
  },

  async saveRule(rule: AutomationRule): Promise<void> {
    return invoke('save_automation_rule', { rule });
  },

  async deleteRule(id: string): Promise<void> {
    return invoke('delete_automation_rule', { id });
  },

  async toggleRule(id: string): Promise<boolean> {
    return invoke('toggle_automation_rule', { id });
  },

  // Config persistence
  async getConfig(): Promise<AppConfig> {
    return invoke('get_app_config');
  },

  async loadConfig(config: AppConfig): Promise<void> {
    return invoke('load_app_config', { config });
  },
};

export default profileService;

File: wsl-tamer-tauri/src-tauri/src/commands/profiles.rs
//! Profile command handlers

use crate::models::{WslProfile, AutomationRule, AppConfig};
use crate::services::ProfileManager;
use std::sync::OnceLock;

/// Global profile manager instance
static PROFILE_MANAGER: OnceLock<ProfileManager> = OnceLock::new();

/// Get or initialize the profile manager
pub fn get_profile_manager() -> &'static ProfileManager {
    PROFILE_MANAGER.get_or_init(|| {
        let manager = ProfileManager::new();
        manager.init_defaults();
        manager
    })
}

/// Get all profiles
#[tauri::command]
pub fn get_profiles() -> Vec<WslProfile> {
    get_profile_manager().get_profiles()
}

/// Get a specific profile
#[tauri::command]
pub fn get_profile(id: String) -> Option<WslProfile> {
    get_profile_manager().get_profile(&id)
}

/// Get current active profile
#[tauri::command]
pub fn get_current_profile() -> Option<WslProfile> {
    get_profile_manager().get_current_profile()
}

/// Save (create/update) a profile
#[tauri::command]
pub fn save_profile(profile: WslProfile) -> Result<(), String> {
    get_profile_manager().save_profile(profile)
}

/// Delete a profile
#[tauri::command]
pub fn delete_profile(id: String) -> Result<(), String> {
    get_profile_manager().delete_profile(&id)
}

/// Set default profile
#[tauri::command]
pub fn set_default_profile(id: String) -> Result<(), String> {
    get_profile_manager().set_default_profile(&id)
}

/// Set current profile and apply to .wslconfig
#[tauri::command]
pub fn apply_profile(id: String) -> Result<(), String> {
    let manager = get_profile_manager();
    
    // Get the profile
    let profile = manager.get_profile(&id)
        .ok_or("Profile not found")?;
    
    // Apply to .wslconfig
    crate::services::WslService::apply_profile(&profile)?;
    
    // Update current profile
    manager.set_current_profile(&id)?;
    
    Ok(())
}

// === Automation Rules ===

/// Get all automation rules
#[tauri::command]
pub fn get_automation_rules() -> Vec<AutomationRule> {
    get_profile_manager().get_rules()
}

/// Save an automation rule
#[tauri::command]
pub fn save_automation_rule(rule: AutomationRule) -> Result<(), String> {
    get_profile_manager().save_rule(rule)
}

/// Delete an automation rule
#[tauri::command]
pub fn delete_automation_rule(id: String) -> Result<(), String> {
    get_profile_manager().delete_rule(&id)
}

/// Toggle automation rule
#[tauri::command]
pub fn toggle_automation_rule(id: String) -> Result<bool, String> {
    get_profile_manager().toggle_rule(&id)
}

/// Get configuration for persistence
#[tauri::command]
pub fn get_app_config() -> AppConfig {
    get_profile_manager().get_config()
}

/// Load configuration from storage
#[tauri::command]
pub fn load_app_config(config: AppConfig) -> Result<(), String> {
    get_profile_manager().load_config(config)
}
